src/Umbraco.Web.UI.Client/src/packages/property-editors/time-zone-picker/property-editor-ui-time-zone-picker.element.ts

```diff
@@ -0,0 +1,145 @@
+import { html, customElement, property, state, css, query } from '@umbraco-cms/backoffice/external/lit';
+import type {
+	UmbPropertyEditorUiElement,
+	UmbPropertyEditorConfigCollection,
+} from '@umbraco-cms/backoffice/property-editor';
+import { UmbLitElement } from '@umbraco-cms/backoffice/lit-element';
+import { UmbChangeEvent } from '@umbraco-cms/backoffice/event';
+import type { UUIRadioEvent } from '@umbraco-cms/backoffice/external/uui';
+import { umbBindToValidation } from '@umbraco-cms/backoffice/validation';
+import type { UmbInputTimeZoneElement } from '@umbraco-cms/backoffice/components';
+
+export interface UmbTimeZonePickerValue {
+	mode: 'all' | 'local' | 'custom';
+	timeZones: Array<string>;
+}
+
+/**
+ * @element umb-property-editor-ui-time-zone-picker
+ */
+@customElement('umb-property-editor-ui-time-zone-picker')
+export class UmbPropertyEditorUITimeZonePickerElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	private _selectedTimeZones: Array<string> = [];
+
+	@state()
+	private _value: UmbTimeZonePickerValue | undefined;
+
+	@property({ type: Object })
+	public set value(value: UmbTimeZonePickerValue | undefined) {
+		this._value = value;
+	}
+
+	public get value(): UmbTimeZonePickerValue | undefined {
+		return this._value;
+	}
+
+	@property({ type: Boolean, reflect: true })
+	readonly: boolean = false;
+
+	@property({ attribute: false })
+	public config?: UmbPropertyEditorConfigCollection;
+
+	@query('umb-input-time-zone')
+	inputElem!: UmbInputTimeZoneElement;
+
+	override connectedCallback() {
+		super.connectedCallback();
+		this._selectedTimeZones = this.value?.timeZones ?? [];
+	}
+
+	#onModeInput(event: UUIRadioEvent) {
+		switch (event.target.value) {
+			case 'all':
+				this.value = { mode: 'all', timeZones: [] };
+				break;
+			case 'local':
+				this.value = { mode: 'local', timeZones: [] };
+				break;
+			case 'custom':
+				this.value = {
+					mode: 'custom',
+					timeZones: Array.from(this.inputElem.value),
+				};
+				break;
+			default:
+				throw new Error(`Unknown value: ${event.target.value}`);
+		}
+
+		this.dispatchEvent(new UmbChangeEvent());
+	}
+
+	#onChange(event: UmbChangeEvent) {
+		const target = event.target as UmbInputTimeZoneElement;
+		const selectedOptions = target.value;
+
+		this._selectedTimeZones = selectedOptions;
```

Comment:
The change from above would make this line unnessecary as the value setter would update this one.